target/linux/ath79/dts/qca9561_tplink_archer-c25-v1.dts

```diff
@@ -0,0 +1,218 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca956x.dtsi"
+
+/ {
+	compatible = "tplink,archer-c25-v1", "qca,qca9560";
+	model = "TP-Link Archer C25 v1";
+
+	chosen {
+		bootargs = "console=ttyS0,115200n8";
+	};
+
+	led_spi {
+		compatible = "spi-gpio";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		gpio-sck = <&gpio 15 GPIO_ACTIVE_HIGH>;
```

Comment:
Word of warning: https://www.kernel.org/doc/Documentation/devicetree/bindings/spi/spi-gpio.txt deprecated (and it looks like they removed?!)  the gpio-sck and gpio-mosi properties in favor of sck-gpios and mosi-gpios.

Reply:
Thanks for the pointer. So this will just be a rename, if I read it correctly:
sck-gpios = <&gpio 15 GPIO_ACTIVE_HIGH>;
mosi-gpios = <&gpio 14 GPIO_ACTIVE_HIGH>;

Do you have any idea how to deal with miso, since I did not find any value and I get
[    0.375991] spi_gpio led_spi: gpio-miso property not found, switching to no-rx mode
in dmesg?
BTW: It is interesting that the message uses the deprecated name "gpio-miso" instead of "miso-gpios".

Reply:
You probably do not need/have a miso pin for you shift-register as only outputs (LEDs) are wired to it.

Reply:
```
[    0.376428] spi_gpio led_spi: gpio-sck property not found
[    0.382022] spi_gpio: probe of led_spi failed with error -2
```
Looks like kernel 4.14 does not know the "new" sck-gpios ...

Reply:
*sight*, 4.19 does:

https://elixir.bootlin.com/linux/v4.19.45/source/Documentation/devicetree/bindings/spi/spi-gpio.txt

The ath79 target already supports 4.19 via the TESTING_KERNEL config. 4.19 knows about both versions and it is not detrimental to specify both.

Reply:
@chunkeey

So you mean I should put:
```
sck-gpios = <&gpio 15 GPIO_ACTIVE_HIGH>;
gpio-sck = <&gpio 15 GPIO_ACTIVE_HIGH>;
mosi-gpios = <&gpio 14 GPIO_ACTIVE_HIGH>;
gpio-mosi = <&gpio 14 GPIO_ACTIVE_HIGH>;
```

?

Reply:
I think so, it's similar to the i2c gpios transition, see for example this:

https://github.com/openwrt/openwrt/blob/aee5c9861996983394ac49f91a831c4641a4bf83/target/linux/ath79/dts/ar7161_aruba_ap-105.dts#L85-L90

Reply:
Okay, I will however wait with another test until eth1/eth0 PR has moved forward.